File: src/main/java/com/wanderers/hotelier_webservices/rest/mapper/BookingMapper.java
package com.wanderers.hotelier_webservices.rest.mapper;

import com.wanderers.hotelier_webservices.rest.model.Booking;
import com.wanderers.hotelier_webservices.server.dto.BookingDto;
import org.springframework.stereotype.Component;

/**
 * Class is responsible for mapping {@link BookingDto} DTO between REST and Server layers
 */
@Component("booking_mapper")
public class BookingMapper {

    public BookingDto mapToBookingDto(Booking booking) {
        return new BookingDto(booking.getAccommodationId(), booking.getCustomerId());
    }

    public Booking mapToRestBooking(BookingDto bookingDto) {
        return new Booking()
                .accommodationId(bookingDto.getAccommodationId())
                .customerId(bookingDto.getCustomerId());
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/rest/mapper/BookingMapper.kt
package com.wanderers.hotelier_webservices.rest.mapper

import com.wanderers.hotelier_webservices.rest.model.Booking
import com.wanderers.hotelier_webservices.server.dto.BookingDto
import org.springframework.stereotype.Component

/**
 * Class is responsible for mapping [BookingDto] DTO between REST and Server layers
 */
@Component("booking_mapper")
class BookingMapper {
    fun mapToBookingDto(booking: Booking): BookingDto {
        return BookingDto(booking.accommodationId, booking.customerId)
    }

    fun mapToRestBooking(bookingDto: BookingDto): Booking {
        return Booking()
            .accommodationId(bookingDto.getAccommodationId())
            .customerId(bookingDto.getCustomerId())
    }
}